### ISSUE TITLE
Dev Test: Ratio and Proportion 1.2.0-dev.31

### ISSUE THREAD

**Issue by zepumph** [phetsims/qa#831]
<h1>Dev Test</h1>

**Mentions**: @emily-phet @terracoda @BLFiedler

<h3>Simulation links</h3>

- [github repository for issues](https://github.com/phetsims/ratio-and-proportion/issues)
- [phet top level directory](https://phet-dev.colorado.edu/html/ratio-and-proportion/1.2.0-dev.31)
- [sim: all_phet.html](https://phet-dev.colorado.edu/html/ratio-and-proportion/1.2.0-dev.31/phet/ratio-and-proportion_all_phet.html)
- [phet-io wrapper index](https://phet-dev.colorado.edu/html/ratio-and-proportion/1.2.0-dev.31/phet-io)
- [sim with camerInput turned on](https://phet-dev.colorado.edu/html/ratio-and-proportion/1.2.0-dev.31/phet/ratio-and-proportion_all_phet.html?cameraInput=hands)
- [sim with camerInput turned on showing video output](https://phet-dev.colorado.edu/html/ratio-and-proportion/1.2.0-dev.31/phet/ratio-and-proportion_all_phet.html?cameraInput=hands&showVideo)

<h3>Test Matrix</h3>

Please include all (non-screen reader) feature testing in these records if applicable.

- [x] Latest macOS, Safari (Time = 4hrs)
- [x] Windows 10, Chrome (Time = 4)
- [x] Windows 10, Firefox (Time = 3 hrs)
- [ ] Latest iOS, Safari (Time = .5)
- [x] Windows 11, Chrome (Time = 4)

Light testing, or optionally skip if time crunch:

- [ ] Latest macOS, Chrome (Time = )
- [x] Latest Chrome OS, Chrome (Time = 1)

If PhET-iO is being tested:

- [x] Latest macOS, Safari (Time = 3.5hrs)
- [ ] Windows 10, Chrome (Time = )
- [ ] Windows 10, Firefox (Time = )
- [x] Latest iOS, Safari (Time = 1)
- [x] Windows 11, Chrome (Time = 3.5)

If screen readers are being tested:

- [x] Win 10/11 Chrome + Jaws (Time = 1.5)
- [ ] Win 10/11 Firefox + NVDA (Time = .5)
- [x] MacOS Safari + VO (Time = 2hrs)
- [ ] iPadOS Mobile VO (Time = .25)

<h3>Features included</h3>

- [x] PhET-iO
- [x] Alternative Input
- [x] UI Sound
- [x] Sonification
- [x] Description
- [x] Voicing
- [x] Camer Input: Hands (see below)

<h3>Focus and Special Instructions</h3>

* This dev test is the moving towards a new publication of Ratio and Proportion that includes Voicing, and Camera Input: Hands (sometimes referred to by its implementation named "Media Pipe").
* Please see https://github.com/phetsims/ratio-and-proportion/issues/488 for instructions on how to test camera input: hands.
* Please focus on Voicing, Camera Input: Hands, and the combination of them. Other features should also be regression tested.
* Let me know if you have any questions!

---
<h2>For QA...</h2>

<details>
<summary><b>General features</b></summary>

<h3>What to Test</h3>

- Click every single button.
- If there is sound, make sure it works.
- Make sure you can't lose anything.
- Play with the sim normally.
- Try to break the sim.
- Try to include browser version numbers
- If there is a console available, check for errors and include them in the Problem Description.
- Run through the string tests on at least one platform, especially if it is about to go to rc.
- Check the Game Up harness on one platform.

<hr>

</details>

<details>
<summary><b>PhET-iO features</b></summary>

<h3>What to Test</h3>

- Make sure that public files do not have password protection. Use a private browser for this.
- Make sure that private files do have password protection. Use a private browser for this.
- Make sure standalone sim is working properly.
- Make sure the wrapper index is working properly.
- Make sure each wrapper is working properly.
- Launch the simulation in Studio with ?stringTest=xss and make sure the sim doesn't navigate to youtube
- For newer PhET-iO wrapper indices, save the "basic example of a functional wrapper" as a .html file and open it. Make
  sure the simulation loads without crashing or throwing errors.

<hr>

</details>

<details>
<summary><b>Accessibility features</b></summary>

<h3>What to Test</h3>

- Specific instructions can be found above.
- Make sure the accessibility (a11y) feature that is being tested doesn't negatively affect the sim in any way. Here is
  a list of features that may be supported in this test:

  - Alternative Input
  - Interactive Description
  - Sound and Sonification
  - Pan and Zoom
  - Mobile Description
  - Voicing
- Test all possible forms of input.
  - Test all mouse/trackpad inputs.
  - Test all touchscreen inputs.
  - Test all keyboard navigation inputs (if applicable).
  - Test all forms of input with a screen reader (if applicable).

<h3>Screen Readers</h3>

This sim may support screen readers. If you are unfamiliar with screen readers, please ask Katie to introduce you to
screen readers. If you simply need a refresher on screen readers, please consult the
[QA Book](https://github.com/phetsims/QA/blob/master/documentation/qa-book.md), which should have all of the information
you need as well as a link to a screen reader tutorial made by Jesse. Otherwise, look over the a11y view before opening
the simulation. Once you've done that, open the simulation and make sure alerts and descriptions work as intended.

<h3>Platforms and Screen Readers to Be Tested</h3>

- Windows 10 + Latest Chrome + Latest JAWS
- Windows 10 + Latest Firefox + Latest NVDA
- macOS + Safari + VoiceOver
- iOS + Safari + VoiceOver (only if specified in testing issue)

<h3>Critical Screen Reader Information</h3>

We are tracking known screen reader bugs in
[here](https://github.com/phetsims/qa/blob/master/documentation/accessibility-bugs.md). If you find a
screen reader bug, please check it against this list.

<h3>Keyboard Navigation</h3>

This sim supports keyboard navigation. Please make sure it works as intended on all platforms by itself and with a
screen reader.

<h3>Magnification</h3>

This sim supports magnification with pinch and drag gestures on touch screens, keyboard shortcuts, and mouse/wheel controls. Please test magnfication and make sure
it is working as intended and well with the use cases of the simulation. Due to the way screen readers handle user input, magnification is NOT expected to
work while using a screen reader so there is no need to test this case.

<hr>

</details>

<details>
<summary><b>FAQs for QA Members</b></summary>

<br>

  

  <details>
  <summary><i>There are multiple tests in this issue... Which test should I do first?</i></summary> 

Test in order! Test the first thing first, the second thing second, and so on.

  </details>

  <br>

  

  <details>
  <summary><i>How should I format my issue?</i></summary>

Here's a template for making issues:

      <b>Test Device</b>

      blah

      <b>Operating System</b>

      blah

      <b>Browser</b>

      blah

      <b>Problem Description</b>

      blah

      <b>Steps to Reproduce</b>

      blah

      <b>Visuals</b>

      blah

      <details>
      <summary><b>Troubleshooting Information</b></summary>

      blah

      </details>

  </details>

  <br>

  

  <details>
  <summary><i>Who should I assign?</i></summary>

We typically assign the developer who opened the issue in the QA repository.

  </details>

  <br>

  

  <details>
  <summary><i>My question isn't in here... What should I do?</i></summary>

You should:

1. Consult the [QA Book](https://github.com/phetsims/QA/blob/master/documentation/qa-book.md).
2. Google it.
3. Ask Katie.
4. Ask a developer.
5. Google it again.
6. Cry.

  </details>

<br>

<hr>

</details>

**Comment by KatieWoe** [comment]
Note: I had mentioned in discussions with some testers that testing should include screen reader + camera input. This is not the case. Testing should include voicing + camera input, as well as screen reader testing on it's own, and the other modes discussed.

**Comment by Nancy-Salpepi** [comment]
@zepumph I am seeing what I reported on the dev-public slack channel on 9/1 with the debug version of the sim. This error appears with mac + safari (not with chrome):
<img width="499" alt="Screen Shot 2022-09-06 at 1 44 42 PM" src="https://user-images.githubusercontent.com/87318828/188704255-e503c5e4-e6c1-4a6a-9689-dde38054508e.png">

I built a version on master and also got the error message there.

**Comment by zepumph** [comment]
What version of safari are you on? Can you create an issue and paste in the slack messages into that issue, assigning me.

**Comment by stemilymill** [comment]
I was playing with the camera input on my personal computer (microsoft surface pro 6, win10 firefox) and noticed that there was significant lag between my hands moving and the sim hands moving. this computer isn't particularly powerful and does have some performance issues, but i thought it was worth mentioning.

**Comment by KatieWoe** [comment]
When I turn voicing on or off in Studio it repeats itself. I.E. it says "Voicing off, voicing off" but it doesn't seem to do so in the main sim, or when the sim is launched. Is it worth an issue?

**Comment by zepumph** [comment]
Yes please.

**Comment by terracoda** [comment]
Is the sim in a wrapper or I frame in studio?

**Comment by KatieWoe** [comment]
Calling QA here

**Comment by zepumph** [comment]
Thanks all! RC imminent.
